rclcpp/include/rclcpp/timer.hpp

```diff
@@ -256,27 +267,28 @@ class GenericTimer : public TimerBase
    * \sa rclcpp::TimerBase::call
    * \throws std::runtime_error if it failed to notify timer that callback will occurr
    */
-  bool
+  std::optional<std::shared_ptr<void>>
   call() override
   {
-    rcl_ret_t ret = rcl_timer_call(timer_handle_.get());
+    rcl_timer_call_info_t timer_call_info_;
```

Comment:
what about creating the shared pointer here and populating it in the line below, to avoid a copy?
```
auto timer_call_info_ = std::make_shared<rcl_timer_call_info_t>();
rcl_ret_t ret = rcl_timer_call_with_info(timer_handle_.get(), timer_call_info_.get());
```

Reply:
I don't know this is a tradeoff thing. The copy is 128 Bytes. But we could avoid a possible not needed memory allocation.

Reply:
I would expect the memory allocation to be needed most of the times, unless someone is cancelling and reactivating the timer at a similar rate as its period

Reply:
changed